utils/preprocessETH.py

```diff
@@ -0,0 +1,117 @@
+        df = unixtime(df)
+        if input_file == 'txtFiles/biwi_hotel.txt': #we realize a rotation 
+            df['pos_x'], df['pos_y'] = -df['pos_y'], df['pos_x']
+
+        n = len(raw_path)
+        df.to_csv(agg_path + input_file[n:-4] + '.csv', index=False)
+
+def find_LU(cfg, agg_path):
+    files = glob.glob(os.path.join(agg_path, '*')) #extract files path  
+    #we specify the amount of cols and rows
+    cols = cfg.MACROPROPS.COLS
+    rows = cfg.MACROPROPS.ROWS
+    #for every file we find the left upper point that maximizes the density
+    for input_file in files:
+        df = pd.read_csv(input_file)
+        minX, minY, maxX, maxY = int(df['pos_x'].min())+1, int(df['pos_y'].min())+1, int(df['pos_x'].max()), int(df['pos_y'].max())
+        iMax, jMax = 0, rows
+        maxCount = 0
+        for i in range(minX, maxX - cols + 1):
+            for j in range(minY + rows, maxY + 1):
+                df_aux = newDF_LU(df, [i, j], cols, rows)
+                if maxCount < df_aux.shape[0]:
+                    maxCount = df_aux.shape[0]
+                    iMax, jMax = i, j
+
+
+        df = newDF_LU(df, [iMax, jMax], cols, rows)  
+        df.to_csv(input_file, index=False) 
+
+def add_vel_angle(agg_path):
+    files = glob.glob(os.path.join(agg_path, '*'))
+
+    #for every file we add two columns: velocity and motion_angle
+    for path_file in files:
+        df = pd.read_csv(path_file)
+        #we transform the position in m to mm
+        df['pos_x'] *= 1000
+        df['pos_y'] *= 1000
+
+        df['vel'] = df['pos_x']
+        df['motion_angle'] = df['pos_x']
+
+        A = [[] for _ in range(int(df['agent_ID'].max() + 1))]
+
+        for i in range(1, len(A)):
+            A[i] = df.loc[df['agent_ID'] == i].index
+
+            if(len(A[i]) == 1): 
+                df = df.loc[df['agent_ID'] != i]
+                continue
+            
+            for j in range(1, A[i].shape[0]):
+                dis = distance(df.loc[A[i][j-1], 'pos_x'], df.loc[A[i][j-1], 'pos_y'],
+                                df.loc[A[i][j], 'pos_x'], df.loc[A[i][j], 'pos_y'])
+                    
+                df.loc[A[i][j], 'motion_angle'] = angle(df.loc[A[i][j-1], 'pos_x'], df.loc[A[i][j-1], 'pos_y'],
+                                                        df.loc[A[i][j], 'pos_x'], df.loc[A[i][j], 'pos_y'])
+                    
+                df.loc[A[i][j], 'vel'] = dis / (df.loc[A[i][j], 'time']-df.loc[A[i][j-1], 'time'])
+
+            if(len(A[i]) > 0):
+                df.loc[A[i][0], 'vel'] = df.loc[A[i][1], 'vel']
+                df.loc[A[i][0], 'motion_angle'] = df.loc[A[i][1], 'motion_angle']     
+
+        df.to_csv(path_file, index=False)        
+
+
+if __name__ == '__main__':
+    cfg = getYamlConfig('config/ETH_ddpm.yml')
+    generate_csv(cfg.DATA_AGGREGATION.RAW_DATA_DIR, cfg.DATA_AGGREGATION.AGG_DATA_DIR)
+    find_LU(cfg, cfg.DATA_AGGREGATION.AGG_DATA_DIR)
+    add_vel_angle(cfg.DATA_AGGREGATION.AGG_DATA_DIR)
```

Comment:
Please use the same convention were we use args first and then default values.
`train.py` has an example of it.